src/cli/agreements.ts

```diff
@@ -0,0 +1,123 @@
+import { flags } from '@oclif/command'
+import { Op } from 'sequelize'
+import LogSymbols from 'log-symbols'
+import Table from 'cli-table3'
+import colors from 'colors/safe'
+
+import BaseCommand from '../utils'
+import Agreement from '../models/agreement.model'
+import { loggingFactory } from '../logger'
+import JobModel from '../models/job.model'
+import { JobState } from '../definitions'
+
+const logger = loggingFactory('cli:agreements')
+
+type AgreementWithJobs = Agreement & { jobs: JobModel[] }
+
+export default class AgreementsCommand extends BaseCommand {
+  static flags = {
+    ...BaseCommand.flags,
+    filter: flags.string({
+      char: 'f',
+      description: 'Filter by status',
+      options: ['active', 'inactive']
+    })
+  }
+
+  static description = 'Agreements info'
+
+  static examples = [
+    '$ rif-pinning agreements',
+    '$ rif-pinning agreements--db myOffer.sqlite',
+    '$ rif-pinning agreements --ls -f active',
+    '$ rif-pinning agreements --ls -f inactive'
+  ]
+
+  static getStatus (agreement: Agreement): string {
+    return agreement.isActive ? LogSymbols.success : LogSymbols.error
+  }
+
+  static getJobInfo (agreement: AgreementWithJobs): string {
+    const [latestJob] = agreement.jobs
+
+    if (!latestJob) return ''
+
+    switch (latestJob.state) {
+      case JobState.ERRORED:
+        return colors.red(`ERRORED(retries: ${latestJob.retry}) - ${latestJob.errorMessage}`)
+      case JobState.FINISHED:
+        return colors.green('PINNED')
+      default:
+        return latestJob.state.toUpperCase()
+    }
+  }
+
+  static expireIn (agreement: Agreement): string {
+    const expired = agreement.expiredIn
+    return expired > 0 ? colors.green(`${expired} min`) : colors.red('EXPIRED')
+  }
+
+  static prepareAgreementForTable (agreement: AgreementWithJobs): any[] {
+    return [
+      AgreementsCommand.getStatus(agreement),
+      `${agreement.agreementReference} ${agreement.dataReference}`,
+      AgreementsCommand.expireIn(agreement),
+      AgreementsCommand.getJobInfo(agreement)
+    ]
+  }
+
+  async queryAgreement (filterStatus?: string): Promise<Array<AgreementWithJobs>> {
+    // TODO write raw query to improve performance
```

Comment:
I don't really think it is required to make performance optimizations for this query. It is run once for CLI command 😉